### DISCUSSION TITLE
vad filter makes whisper combine different segments into one line

### DISCUSSION THREAD

**Discussion by phwang4** [SYSTRAN/faster-whisper#1015] [Q&A]
I'm trying to transcribe this audio file where the speaker says a sentence at the very beginning and then again at the very end (about 420 seconds). 
I use the vad filter so that whisper doesn't hallucinate a bunch of Thank You's every 30 seconds or so of silence, but it's combining the first and last sentence into one chunk and giving me an output that looks like so:
3s -> 420s: Sentence1. Sentence2

I want it to look like this:
3s -> 4s: Sentence 1.
419s -> 420s: Sentence 2.

Any ideas on how to improve this? 
I've tried using word_timestamps and hallucination_silence_threshold instead of the silence filter but that always leads to some kind of repetitive hallucinations on silence. I've also tried messing with the vad_parameters but I can't seem to get it to split.

**Comment by Hikari-Tsai** [comment]
This happens because VAD removes silence and the transcription model sees the retained speech chunks concatenated together. The timestamp-restoration step maps the generated segment's start to the first original chunk and its end to the last one; if one decoded segment spans both chunks, its restored range can therefore cover the long silence between them.

VAD controls which audio reaches Whisper, not subtitle/sentence segmentation. For reliable separate cues, enable word timestamps and split the output whenever adjacent words have a large original-time gap:

```python
segments, info = model.transcribe(
    audio,
    vad_filter=True,
    word_timestamps=True,
    vad_parameters={"min_silence_duration_ms": 500},
)

for segment in segments:
    words = segment.words or []
    # Start a new subtitle whenever word.start - previous_word.end
    # exceeds your chosen gap, e.g. 1.0 second.
```

Alternatively, call `get_speech_timestamps` first and transcribe each returned speech chunk separately (or pass explicit `clip_timestamps`). That guarantees that Whisper cannot create one text segment spanning two distant speech regions.

Changing only the VAD probability threshold will not guarantee subtitle boundaries. `min_silence_duration_ms` changes VAD chunking, while final display segmentation should be handled from VAD/word timestamps according to your subtitle requirements.
